allennlp/data/vocabulary.py

```diff
@@ -460,6 +460,7 @@ def _extend(self,
         counter = counter or {}
         tokens_to_add = tokens_to_add or {}
 
+        self._retained_counter = counter
```

Comment:
Please can you set this attribute to `None` in __init__ before you use it? We broadly try to only modify attributes in methods, rather than set them.

Reply:
Sure 👍